### PR TITLE
Edition2018

### PR DESCRIPTION
**Pull request by tafia**
- style: convert to rust edition 2018
- feat: add `Writer::write_nested_event`
- feat: add `BytesStart::try_get_attribute`
- test: add more test on github actions

### CODE REVIEW

File: src/writer.rs [hunk 1]
```diff
@@ -170,6 +168,22 @@ impl<W: Write> Writer<W> {
...
+        &mut self,
+        parent_name: N,
+        event: E,
+    ) -> Result<()> {
+        let name = parent_name.as_ref();
+        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
+        self.write_event(event)?;
+        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
+        Ok(())
+    }
```
Comment (dralley) [inline src/writer.rs:186]: :ok_hand: :+1:

Comment (dralley) [reply src/writer.rs:186]: Is there any way to allow setting attributes on the start tag without hurting the API too much? e.g. to support sequences like

```  <checksum type="sha256">47bb3f2a77d01bd38e462765ebd67f8890af8c4ea75ab1b9ba192926db7e552b</checksum>
```

Comment (dralley) [inline src/writer.rs:186]: Would it be possible to allow setting attributes without making it too difficult to use? e.g.

```xml
  <checksum type="sha256">fc83446cfb15634f0200a29a0138e22a011ea4a064d531c0c53402b56a46ccab</checksum>
```

An API that emulates the [Entry API](https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html) for hash maps might be nice (but probably a lot more work).  Maybe something like this:

```rust
writer.create_node("checksum")
      .push_attribute(("type", "sha256")
      .nested_event(Event::Text(...));
```

Comment (tafia) [reply src/writer.rs:186]: That's a good suggestion let me try

Comment (dralley) [reply src/writer.rs:186]: Are you currently working on it?  If not, I could give it a try

Comment (tafia) [reply src/writer.rs:186]: Sorry I got something more urgent to do at work. Feel free to give it a try I'll review.
Thanks!

Comment (dralley) [reply src/writer.rs:186]: @tafia I gave it a shot, let me know what you think.  It's definitely not perfect but I'm posting it so I can go to sleep :)

https://github.com/tafia/quick-xml/pull/278

**Comment by dralley** [comment]
@tafia Do you have any plans for a new release soon?

**Comment by dralley** [comment]
@tafia ^

**Comment by dralley** [comment]
It would be good to at least merge this PR to avoid conflicts with any other PRs

**Pull request merged**